bokehjs/src/coffee/document.coffee

```diff
@@ -23,7 +24,10 @@ export class ModelChangedEvent extends DocumentChangedEvent
       throw new Error("'id' field should never change, whatever code just set it is wrong")
 
     value = @new_
-    value_json = HasProps._value_to_json('new_', value, @model)
+    if @model instanceof HasProps
+      value_json = @model.constructor._value_to_json(@attr, value, @model)
+    else
+      value_json = HasProps._value_to_json(@attr, value, @model)
```

Comment:
I'm not sure I understand what's going on above... aren't all models instances of HasProps?
It feels like this block of code should be encapsulated in HasProps._value_to_json somehow.